### PR TITLE
update path in title when updating directories related to study

### PR DESCRIPTION
**Pull request by anas-drissi**

### CODE REVIEW

**Review by bendaamerahm — CHANGES_REQUESTED** [review]
that work fine but we never use directory notifications directly into gridstudy (to avoid mixed context) , there's a solution using an existing endpoint in study server , we can call it to invoke notification coming from outside.
=>  **this comment is outdated  : now we can use directory notifications directly into gridstudy :+1:**

File: src/components/study-container.js [hunk 1]
```diff
@@ -382,36 +385,70 @@ export function StudyContainer({ view, onChangeTab }) {
...
+            console.error('Unexpected Notification WebSocket error', event);
+        };
+        // We must save wsRef.current in a variable to make sure that when close is called it refers to the same instance.
+        // That's because wsRef.current could be modify outside of this scope.
+        const wsToClose = wsRef.current;
+        // cleanup at unmount event
+        return () => {
+            wsToClose.close();
+        };
+    }, [fetchStudyPath, studyUuid]);
```
Comment (sBouzols) [inline src/components/study-container.js:444]: we want the websocket creation at mount event then we must have 0 dependencies here for this effect.

File: src/utils/rest-api.js [hunk 2]
```diff
@@ -37,6 +39,30 @@ function backendFetch(url, init) {
...
+        webSocketBaseUrl +
+        PREFIX_DIRECTORY_NOTIFICATION_WS +
+        '/notify?updateType=directories';
+
+    const reconnectingWebSocket = new ReconnectingWebSocket(
+        () => webSocketUrl + '&access_token=' + getToken()
+    );
+    reconnectingWebSocket.onopen = function (event) {
+        console.info(
+            'Connected Websocket update studies ' + webSocketUrl + ' ...'
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/utils/rest-api.js]: it's fixed

File: src/components/study-container.js (hunk 1 above)
Comment (anas-drissi) [reply src/components/study-container.js]: it's fixed

File: src/components/study-container.js [hunk 3]
```diff
@@ -382,36 +385,70 @@ export function StudyContainer({ view, onChangeTab }) {
...
+        wsRef.current = connectNotificationsWsUpdateDirectories();
+
+        wsRef.current.onmessage = function (event) {
+            const eventData = JSON.parse(event.data);
+            if (eventData.headers) {
+                if (
+                    eventData.headers['notificationType'] ===
+                    directoriesNotificationType.UPDATE_DIRECTORY
+                ) {
+                    fetchStudyPath(studyUuid);
```
[the remark this answers is not in the document]
Comment (anas-drissi) [reply src/components/study-container.js:468]: We filter this through eventData.headers['notificationType'] === directoriesNotificationType.UPDATE_DIRECTORY to retrieve the new study path "fetchStudyPath(studyUuid)".

**Review by jonenst — APPROVED** [review]

**Review by jonenst — APPROVED** [review]

**Pull request merged**